### PR TITLE
Upload with thirdweb storage-server gateway

### PR DESCRIPTION
**Pull request by nessup**

### CODE REVIEW

File: packages/storage/test/ipfs.test.ts [hunk 1]
```diff
@@ -442,12 +448,14 @@ describe("IPFS", async () => {
...
     );
 
-    expect(uris[0]).to.equal(
-      "ipfs://QmTtEY2WSTDpzYSXw2G3xsYw3eMs8YephvrfVYd8qia9F9/0",
-    );
-    expect(uris[1]).to.equal(
-      "ipfs://QmTtEY2WSTDpzYSXw2G3xsYw3eMs8YephvrfVYd8qia9F9/1",
-    );
+    expect(uris[0]).to.be.oneOf([
+      "ipfs://QmTtEY2WSTDpzYSXw2G3xsYw3eMs8YephvrfVYd8qia9F9/0", // using Pinata
```
Comment (jakeloo) [inline packages/storage/test/ipfs.test.ts:452]: cidv0

File: packages/storage/test/ipfs.test.ts [hunk 2]
```diff
@@ -442,12 +448,14 @@ describe("IPFS", async () => {
...
+      "ipfs://bafybeibvqqvjjdxmhnzru7y7b35smpveu7j75hfgfgsssosy7mzkhrvhjq/0" // using web3.storage
```
Comment (jakeloo) [inline packages/storage/test/ipfs.test.ts:453]: cidv1

File: packages/storage/src/common/urls.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { GatewayUrls } from "../types";
 export const DEFAULT_GATEWAY_URLS: GatewayUrls = {
   // Note: Gateway URLs should have trailing slashes (we clean this on user input)
   "ipfs://": [
+    // "http://127.0.0.1:8787/ipfs/", // TODO: Remove this before merging to main
```
Comment (joaquim-verges) [inline packages/storage/src/common/urls.ts:9]: remove

File: packages/storage/src/core/uploaders/ipfs-uploader.ts [hunk 4]
```diff
@@ -308,8 +353,11 @@ export class IpfsUploader implements IStorageUploader<IpfsUploadBatchOptions> {
         return reject(new Error("Unknown upload error occured"));
       });
 
-      xhr.open("POST", PINATA_IPFS_URL);
-      xhr.setRequestHeader("Authorization", `Bearer ${token}`);
+      xhr.open("POST", ipfsUploadUrl);
+
+      if (uploadToken) {
+        xhr.setRequestHeader("Authorization", `Bearer ${uploadToken}`);
```
Comment (joaquim-verges) [inline packages/storage/src/core/uploaders/ipfs-uploader.ts:359]: @nessup do you think we should be passing the default API key here in preparation?

**Review by joaquim-verges — APPROVED** [review]
looks good, I think we can merge. just a couple of comments

File: packages/storage/src/common/constants.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const DEFAULT_API_KEY =
+  "fb52f0ca630bdd02f0d5210d9e8e3568272ac765dd0709272918eefb29d3ac9e";
```
Comment (joaquim-verges) [inline packages/storage/src/common/constants.ts:2]: remember to add this key to https://www.notion.so/thirdweb/Universal-RPC-API-keys-4634982244164fc4a51962e9e1ae3861

**Review by joaquim-verges — APPROVED** [review]
lg!

**Comment by joaquim-verges** [comment]
closing for now

**Pull request closed without merging**